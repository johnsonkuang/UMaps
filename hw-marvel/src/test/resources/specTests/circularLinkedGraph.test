# Tests that on a small-scale graph that all operations work, including loading from a tsv and BFS

LoadGraph graph circularLinkedGraph.tsv

ListChildren graph 1
ListChildren graph 2
ListChildren graph 3
ListChildren graph 4

FindPath graph 1 3

FindPath graph 2 4

FindPath graph 3 1

FindPath graph 4 2

ListNodes graph