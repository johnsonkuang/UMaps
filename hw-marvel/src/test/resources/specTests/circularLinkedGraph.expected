# Tests that on a small-scale graph that all operations work, including loading from a tsv and BFS

loaded graph graph

the children of 1 in graph are: 2(A) 4(D)
the children of 2 in graph are: 1(A) 3(B)
the children of 3 in graph are: 2(B) 4(C)
the children of 4 in graph are: 1(D) 3(C)

path from 1 to 3:
1 to 2 via A
2 to 3 via B

path from 2 to 4:
2 to 1 via A
1 to 4 via D

path from 3 to 1:
3 to 2 via B
2 to 1 via A

path from 4 to 2:
4 to 1 via D
1 to 2 via A

graph contains: 1 2 3 4